pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -894,55 +895,61 @@ func (r *Builder) PopulatorVolumes(vmRef ref.Ref, annotations map[string]string,
 	}
 
 	for _, image := range images {
-
 		if image.Status != string(ImageStatusActive) {
 			r.Log.Info("the image is not ready yet", "image", image.Name)
 			continue
 		}
-
-		originalVolumeDiskId := image.Name
-		if imageProperty, ok := image.Properties[forkliftPropertyOriginalVolumeID]; ok {
-			originalVolumeDiskId = imageProperty.(string)
-		}
-
-		_, err = r.getVolumePopulator(image.Name)
+		var pvc *core.PersistentVolumeClaim
+		pvc, err = r.ensureVolumePopulator(workload, &image, annotations, secretName)
 		if err != nil {
-			if !k8serr.IsNotFound(err) {
-				err = liberr.Wrap(err)
+			if !k8serr.IsAlreadyExists(err) {
+				err = liberr.Wrap(err, "couldn't build the PVC",
+					"image", image.Name)
 				return
 			}
+			err = nil
+			continue
 		}
+		pvcNames = append(pvcNames, pvc.Name)
+	}
+	return
+}
 
+func (r *Builder) ensureVolumePopulator(workload *model.Workload, image *model.Image, annotations map[string]string, secretName string) (pvc *core.PersistentVolumeClaim, err error) {
+	populator, err := r.getVolumePopulator(image.Name)
+	if err != nil {
+		if !k8serr.IsNotFound(err) {
+			err = liberr.Wrap(err)
+			return
+		}
 		var populatorName string
-		populatorName, err = r.createVolumePopulatorCR(image, secretName, vmRef.ID)
+		populatorName, err = r.createVolumePopulatorCR(*image, secretName, workload.ID)
 		if err != nil {
 			err = liberr.Wrap(err)
 			return
 		}
+		pvc, err = r.ensurePopulatorPVC(workload, image, annotations, populatorName, secretName)
+		return
+	}
+	err = k8serr.NewAlreadyExists(schema.ParseGroupResource("openstackvolumepopulator.forklift.konveyor.io"), populator.Name)
```

Comment:
I'd expect 'ensure' method to return err == nil when the entity already exists - this behavior is surprising

Reply:
Yeah maybe that's better approach, but then we would need to check if the returned pvc is `nil` or not

Reply:
Please take look now, I think it's better now.